src/lib/orionld/common/orionldState.h

```diff
@@ -25,6 +25,22 @@
 *
 * Author: Ken Zangelin
 */
+
+//
+// FIXME: Regardless of what DB is selected in dbConfiguration.h, I need to include "mongo/client/dbclient.h"
+//        This is this way because mongoBackend/MongoGlobal.cpp (entitiesQuery) uses orionldState.qMongoFilterP for the new NGSI-LD Q-filters.
+//        mongoBackend will ALWAYS use the old C++ Legacy driver so not sure this will ever be "fixed".
+//        At least not while we still use mongoBackend - the idea is to one day stop using mongoBackend
+//
+/* My branch changes LARYSSE
+#include "mongo/client/dbclient.h"                               // MongoDB C++ Client Legacy Driver
+
+#include "orionld/db/dbConfiguration.h"                          // DB_DRIVER_MONGOC, or not ...
+#include "orionld/db/dbDriver.h"                                 // database driver header
+
+#ifdef DB_DRIVER_MONGOC
+#include "mongoc/mongoc.h"                                       // MongoDB C Client Driver - if needed
+#endif*/
```

Comment:
This seems like a mix between my own changes and yours, Larysse ...
First of all, to comment away a block, use `#if 0 / #endif` instead of /* */, it's much clearer where it starts/ends.

After that side-comment, how do we need to leave this int he PR?

Looking at dbDriver.h, the whole `#ifdef DB_DRIVER_MONGOC` is already done there so, I think it's safe to remove the part that you commented out.

Reply:
Fixed in e8fc9cb